Implement separate results for `rstrnt-report-result` (#2689)
teemtee/tmt: docs/releases.rst

:ref:`beaker.pool</spec/hardware/beaker>` hardware requirements.

> tmt-1.34

Reviewer: 1.35 hopefully :)

Reviewer 2: Addressed in 1a3d13c7

The linting of tests, plans and stories has been extended by detecting
duplicate ids.

The ``tmt-report-result`` command has been updated to generate a
``results.yaml`` file with content which differ from the

> previously generated ``restraint-result`` file.

Reviewer 2: ```suggestion
previously generated ``restraint-result`` file. New ``results.yaml``
follows :ref:`/spec/plans/results` specification.
```

The behaviour for existing tests which already utilise the
``tmt-report-result`` command shall remain unchanged, in that the
overall result returned by tmt shall be determined by parsing the
``results.yaml`` file and selecting the result property with the
value which resides highest on the hierarchy of ``skip``, ``pass``,
``warn``, ``fail``.

> By setting ``result: custom`` in the ``fmf`` file the results.yaml

Reviewer 3: I would make it more concrete, we use `fmf` files for plans, tests and stories, so in `tmt test metadata`

> By setting ``result: custom`` in the ``fmf`` file the results.yaml

Reviewer 3: make it concrete in which metadata, this seems to be a new attribute in the test metadata?

Reviewer 2: ```suggestion
By setting :ref:`result key</spec/tests/result>` of a test to ``custom`` the ``results.yaml``
```
